packages/devtools_app/lib/src/debugger/codeview.dart

```diff
@@ -243,81 +244,91 @@ class _CodeViewState extends State<CodeView>
 
     _updateScrollPosition(animate: false);
 
-    return OutlineDecoration(
-      child: Column(
-        children: [
-          buildCodeviewTitle(theme),
-          if (lines.isNotEmpty)
-            DefaultTextStyle(
-              style: theme.fixedFontStyle,
-              child: Expanded(
-                child: Scrollbar(
-                  controller: textController,
-                  child: ValueListenableBuilder<StackFrameAndSourcePosition>(
-                    valueListenable: widget.controller.selectedStackFrame,
-                    builder: (context, frame, _) {
-                      final pausedFrame = frame == null
-                          ? null
-                          : (frame.scriptRef == scriptRef ? frame : null);
-
-                      return Row(
-                        children: [
-                          ValueListenableBuilder<
-                              List<BreakpointAndSourcePosition>>(
-                            valueListenable:
-                                widget.controller.breakpointsWithLocation,
-                            builder: (context, breakpoints, _) {
-                              return Gutter(
-                                gutterWidth: gutterWidth,
-                                scrollController: gutterController,
-                                lineCount: lines.length,
+    return HistoryViewport(
+      history: widget.controller.scriptsHistory,
+      generateTitle: (script) => script.uri,
+      controls: [
+        ScriptPopupMenu(widget.controller),
+        ScriptHistoryPopupMenu(
+          itemBuilder: _buildScriptMenuFromHistory,
+          onSelected: (scriptRef) {
+            widget.controller.showScriptLocation(ScriptLocation(scriptRef));
+          },
+          enabled: widget.controller.scriptsHistory.hasScripts,
+        ),
+      ],
+      buildContents: (context, script) {
```

Comment:
super-nit: can you rename to `contentBuilder` to match naming conventions for other builders in flutter